pinot-core/src/main/java/org/apache/pinot/core/io/writer/impl/FixedByteSingleValueMultiColWriter.java

```diff
@@ -101,7 +101,9 @@ public void setBytes(int row, int col, byte[] bytes) {
 
   public void close()
       throws IOException {
-    this.indexDataBuffer.close();
-    this.indexDataBuffer = null;
+    if (indexDataBuffer != null) {
```

Comment:
indexDataBuffer can never be null. You might want to change all the member variables as final for this class

Reply:
Done. I introduced the null check to make the close() is idempotent. But since the close() on PinotDataBuffer is idempotent, calling this close() multiple times will be a NO-OP. 
Made the member variables final. rowSizeInBytes can't be final as it is being incremented in the constructor.